blob/service.go

```diff
@@ -354,14 +354,150 @@ func (s *Service) retrieve(
 		return nil, nil, err
 	}
 
-	eds, err := s.shareGetter.GetEDS(ctx, header)
+	headerGetterSpan.SetStatus(codes.Ok, "")
+	headerGetterSpan.AddEvent("received eds", trace.WithAttributes(
+		attribute.Int64("eds-size", int64(len(header.DAH.RowRoots)))))
+
+	rowIndex := -1
+	for i, row := range header.DAH.RowRoots {
+		if !namespace.IsOutsideRange(row, row) {
+			rowIndex = i
+			break
+		}
+	}
+
```

Comment:
Should we verify here that `rowIndex` is not `-1` ?

Reply:
[b241e7e](https://github.com/celestiaorg/celestia-node/pull/3633/commits/b241e7e8171ac1dc18dfea63c4a2b88fa905d722)